src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java

```diff
@@ -175,4 +199,30 @@ ShutdownReason getReason() {
         return reason;
     }
 
+    private boolean isShardInContextParentOfAny(List<Shard> shards) {
+        for(Shard shard : shards) {
+            if (isChildShardOfShardInContext(shard)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean isChildShardOfShardInContext(Shard shard) {
+        return (StringUtils.equals(shard.getParentShardId(), shardInfo.getShardId())
+                || StringUtils.equals(shard.getAdjacentParentShardId(), shardInfo.getShardId()));
+    }
+
+    private void dropLease() {
```

Comment:
Looking into this again, why can't we use `leaseManager.getLease(shardInfo.getShardId())` here?

Reply:
Using leaseCoordinator.getCurrentlyHeldLease now